predicates/forwarded/forwarded_test.go

```diff
@@ -0,0 +1,308 @@
+	headers http.Header
+}
+
+func TestForwardedHost(t *testing.T) {
+	testCases := []struct {
+		msg     string
+		host    string
+		r       request
+		matches bool
+		isError bool
+	}{{
+		msg:     "Empty host should fail",
+		host:    "",
+		r:       request{},
+		matches: false,
+		isError: true,
+	}, {
+		msg:     "Empty Forwarded Header should not match",
+		host:    "example.com",
+		r:       request{},
+		matches: false,
+		isError: false,
+	}, {
+		msg:  "Same forwarded Header should match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"for=192.0.2.60;proto=http;by=203.0.113.43;host=example.com"},
+			},
+		},
+		matches: true,
+		isError: false,
+	}, {
+		msg:  "Same forwarded Header should match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"for=192.0.2.60;proto=http;host=example.com;by=203.0.113.43"},
+			},
+		},
+		matches: true,
+		isError: false,
+	}, {
+		msg:  "Same forwarded Header should match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"for=192.0.2.60;host=example.com;proto=http;by=203.0.113.43"},
+			},
+		},
+		matches: true,
+		isError: false,
+	}, {
+		msg:  "Same forwarded Header should match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"host=example.com;for=192.0.2.60;proto=http;by=203.0.113.43"},
+			},
+		},
+		matches: true,
+		isError: false,
+	}, {
+		msg:  "Different forwarded Header should not match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"for=192.0.2.60;proto=http;by=203.0.113.43;host=example.org"},
+			},
+		},
+		matches: false,
+		isError: false,
+	}, {
+		msg:  "Forwarded Header with no host should not match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"for=192.0.2.60;proto=http;by=203.0.113.43"},
+			},
+		},
+		matches: false,
+		isError: false,
+	}, {
+		msg:  "Forwarded Header only host should match",
+		host: "example.com",
+		r: request{
+			url: "https://myproxy.com/index.html",
+			headers: http.Header{
+				"Forwarded": []string{"host=example.com"},
+			},
+		},
+		matches: true,
+		isError: false,
+	}}
+
+	for _, tc := range testCases {
+
+		t.Run(tc.msg, func(t *testing.T) {
+			spec := NewForwardedHost()
+
+			p, err := spec.Create([]interface{}{tc.host})
+			hasError := err != nil
+			if hasError || tc.isError {
+				if !tc.isError {
+					t.Error("Predicate creation failed")
+				}
+
+				if !hasError {
+					t.Error("Predicate should have failed")
+				}
+
+				return
+			}
+
+			r, err := newRequest(tc.r)
+			if !tc.isError && err != nil {
```

Comment:
I think `!tc.isError` is not needed here. Also if we can not create the request it makes sense to fail with Fatal.